quickevent/app/quickevent/plugins/Event/src/services/emmaclient.cpp

```diff
@@ -267,48 +267,54 @@ void EmmaClient::exportFinish()
 	QTextStream ts(&f);
 
 	int current_stage = getPlugin<EventPlugin>()->currentStageId();
-	qfs::QueryBuilder qb;
-	qb.select2("cards", "id, siId")
-            .select2("runs", "finishTimeMs, misPunch, badCheck, disqualified, notCompeting")
-			.from("cards")
-			.join("cards.runId", "runs.id")
-			.where("cards.stageId=" QF_IARG(current_stage))
-            .orderBy("cards.id ASC");
 	int start00 = getPlugin<EventPlugin>()->stageStartMsec(current_stage);
 
+	qfs::QueryBuilder qb;
 	qfs::Query q2;
+
+	qb.select2("runs", "siId, isRunning, finishTimeMs, misPunch, badCheck, disqualified, notCompeting")
+			.from("runs")
+			.where("runs.stageId=" QF_IARG(current_stage))
+			.where("runs.siId IS NOT NULL")
```

Comment:
Wasn't be sufficient to filter out runners who do not have check time and are running?

Something like:
`.where("runs.checkTimeMs IS NOT NULL OR runs.isRunning IS NULL")`
or
`.where("runs.checkTimeMs IS NOT NULL OR NOT runs.isRunning")`
(I don't know which type is `isRunning` in DB)

Reply:
I cannot filter out `isRunning`, because of DNS info.
Better than `checkTimeMs` si check if there is record in `cards`.